clang/lib/Sema/SemaDeclAttr.cpp

```diff
@@ -2857,32 +2857,64 @@ static void handleWeakImportAttr(Sema &S, Decl *D, const ParsedAttr &AL) {
 }
 
 // Checks correctness of mutual usage of different work_group_size attributes:
-// reqd_work_group_size, max_work_group_size. Values of reqd_work_group_size
-// arguments shall be equal or less than values coming from max_work_group_size.
+// reqd_work_group_size, max_work_group_size and max_global_work_dim.
+// Values of reqd_work_group_size arguments shall be equal or less than values
+// coming from max_work_group_size.
+// In case of 'max_global_work_dim' attribute equals to 0 we shall ensure
+// that if max_work_group_size and reqd_work_group_size attributes exist,
+// then they are equal (1, 1, 1).
 static bool checkWorkGroupSizeValues(Sema &S, Decl *D, const ParsedAttr &Attr,
                                      uint32_t WGSize[3]) {
+  bool Result = true;
+  if (Attr.getKind() == ParsedAttr::AT_SYCLIntelMaxGlobalWorkDim) {
+    if (const SYCLIntelMaxWorkGroupSizeAttr *A =
```

Comment:
If worth applying, this suggestion goes for all of the similar conditions:
```suggestion
    if (const auto *A =
```

Reply:
Done.